### PR TITLE
Full documentation for the metric control

### PR DESCRIPTION
**Pull request by FabienLelaquais**
Addresses #975.

### CODE REVIEW

File: docs/manuals/userman/gui/viselements/generic/metric.md_template [hunk 1]
```diff
@@ -1,14 +1,272 @@
...
+
+## Setting the gauge type {data-source="gui:doc/examples/controls/metric-type.py"}
+
+You can change the representation to an horizontal bar by setting the [*type*](#p-type) 
+
+Here is the definition of a `metric` control that is horizontal:
+!!! taipy-element
+    default={value}
+    delta={delta}
+    threshold={threshold}
```
Comment (FredLL-Avaiga) [inline docs/manuals/userman/gui/viselements/generic/metric.md_template:72]: where is the type parameter ?

File: docs/manuals/userman/gui/viselements/generic/chat.md_template [hunk 2]
```diff
@@ -164,7 +164,7 @@ The control's `on_action` callback property is set so that the callback function
 invoked when the "Submit" button is pressed.<br/>
 Here is the code of this function:
 ```python linenums="1"
-users: list[str|Icon] = []
+users: list[str | Icon] = []
```
Comment (namnguyen20999) [inline docs/manuals/userman/gui/viselements/generic/chat.md_template]: I don't think this syntax will be support for python 3.8 as @FredLL-Avaiga mentioned

File: docs/manuals/userman/gui/viselements/generic/metric.md_template [hunk 3]
```diff
@@ -1,14 +1,272 @@
...
+<figure class="tp-center">
+    <img src="../metric-formats-d.png" class="visible-dark"  width="80%"/>
+    <img src="../metric-formats-l.png" class="visible-light" width="80%"/>
+    <figcaption>Formatted values</figcaption>
+</figure>
+
+## Changing the indicator color {data-source="gui:doc/examples/controls/metric-delta_color.py"}
+
+By default, a positive value [*delta*](#p-delta) is rendered in green, indicating a good thing.
+Negative values are displayed in red, hinting at a bad thing.<br/>
```
Comment (FredLL-Avaiga) [inline docs/manuals/userman/gui/viselements/generic/metric.md_template:157]: good vs bad :-)

**Review by FredLL-Avaiga — APPROVED** [review]

**Review by namnguyen20999 — APPROVED** [review]

File: docs/manuals/userman/gui/viselements/generic/metric.md_template (hunk 3 above)
Comment (FabienLelaquais) [reply docs/manuals/userman/gui/viselements/generic/metric.md_template:157]: Yeah. Evil was a bit too strong.

File: docs/manuals/userman/gui/viselements/generic/chat.md_template (hunk 2 above)
Comment (FabienLelaquais) [reply docs/manuals/userman/gui/viselements/generic/chat.md_template]: Correct

File: docs/manuals/userman/gui/viselements/generic/metric.md_template (hunk 1 above)
Comment (FabienLelaquais) [reply docs/manuals/userman/gui/viselements/generic/metric.md_template:72]: That's right. Something went very wrong here.
On it. Thanks.

File: docs/manuals/userman/gui/viselements/generic/chat.md_template (hunk 2 above)
Comment (FabienLelaquais) [reply docs/manuals/userman/gui/viselements/generic/chat.md_template]: Fixed

**Review by FredLL-Avaiga — APPROVED** [review]

File: docs/manuals/userman/gui/viselements/generic/metric.md_template [hunk 4]
```diff
@@ -1,14 +1,274 @@
...
+Renders as such:
+<figure class="tp-center">
+    <img src="../metric-simple-d.png" class="visible-dark"  width="80%"/>
+    <img src="../metric-simple-l.png" class="visible-light" width="80%"/>
+    <figcaption>Simple case</figcaption>
+</figure>
+
+## Setting the gauge type {data-source="gui:doc/examples/controls/metric-type.py"}
+
+You can change the representation to an horizontal bar by setting the [*type*](#p-type) property to
```
Comment (dinhlongviolin1) [inline docs/manuals/userman/gui/viselements/generic/metric.md_template]: an -> a

**Review by dinhlongviolin1 — APPROVED** [review]
LGTM

**Review by FlorianJacta — APPROVED** [review]
I like it!

File: docs/manuals/userman/gui/viselements/generic/metric.md_template (hunk 4 above)
Comment (FabienLelaquais) [reply docs/manuals/userman/gui/viselements/generic/metric.md_template]: Thanks! I always miss this one.

**Review by dinhlongviolin1 — APPROVED** [review]

**Pull request merged**